### PR TITLE
Client-side logging capture

### PR DESCRIPTION
**Pull request by hiddentao**
This captures `console.log` and related calls in the Mist UI layer and sends them via IPC to the backend log4js logging system. 

This PR also fixes a minor bug related to loading and playing UI sounds client-side.

**Note:** The Mist UI currently crashes me after loading due to some other unknown bug (other users are seeing it too - #472). Perhaps I can fix it as part of this PR.

### CODE REVIEW

**Comment by hiddentao** [comment]
Output looks something like:

```
[2016-04-19 17:11:47.617] [INFO] checkNodeSync - No sync necessary, starting app!
[2016-04-19 17:11:47.618] [INFO] main - Loading Interface at http://localhost:3000
(electron) getUrl is deprecated. Use getURL instead.
[2016-04-19 17:11:48.679] [DEBUG] (ui: mist) - (ipc callback from window with url: http://localhost:3000/)
[2016-04-19 17:11:48.679] [INFO] (ui: mist) - Meteor starting up...
[2016-04-19 17:11:48.987] [INFO] (ui: mist) - http://about:blank
```

File: interface/client/appStart.js [hunk 1]
```diff
@@ -41,6 +39,27 @@ mistInit = function(){
 
 
 Meteor.startup(function(){
+    // send console logging to IPC backend
```
Comment (frozeman) [inline interface/client/appStart.js]: Thats exactly what should be as a module in a preloaded file ;)so that not every dapp needs to have that and we can watch all dapps logs, if we wanted to

File: interface/client/mistAPIBackend.js [hunk 2]
```diff
@@ -17,15 +19,7 @@ var filterId = function(str) {
 };
 
 
-var sound = {
-    bip: document.createElement('audio'),
-    bloop: document.createElement('audio'),
-    invite: document.createElement('audio'),
-};
-sound.bip.src = 'file://'+ dirname + '/../../sounds/bip.mp3';
```
Comment (frozeman) [inline interface/client/mistAPIBackend.js]: I did it like this, so that the sounds are preloaded. With your approach it would load the file in the moment it uses them, which could lead to delays, i guess.

File: modules/ipc/ipcProviderBackend.js [hunk 3]
```diff
@@ -593,4 +597,24 @@ module.exports = function(){
     ipc.on('ipcProvider-writeSync', function(event, payload){
         sendRequest(event, payload, true);
     });
-};
\ No newline at end of file
+
+    const uiLoggers = {};
+
+    ipc.on('console_log', function(event, id, logLevel, logItemsStr) {
```
Comment (frozeman) [inline modules/ipc/ipcProviderBackend.js]: This whole stuff belongs not here, but in: https://github.com/ethereum/mist/blob/master/modules/ipcCommunicator.js

File: interface/client/mistAPIBackend.js (hunk 2 above)
Comment (hiddentao) [inline interface/client/mistAPIBackend.js]: This was throwing errors in the JS console on Mist startup, because `dirname` is not available in the Meteor client-side. Neither is `__dirname` or any of the standard Node.js stuff, though Meteor 1.3 probably has support for it. In any case, since the file is being loaded from the local FS delays should be minimal.

File: interface/client/appStart.js (hunk 1 above)
Comment (hiddentao) [inline interface/client/appStart.js]: Yeah, good idea.

File: modules/ipc/ipcProviderBackend.js (hunk 3 above)
Comment (hiddentao) [inline modules/ipc/ipcProviderBackend.js]: Will move.

File: interface/client/mistAPIBackend.js (hunk 2 above)
Comment (frozeman) [inline interface/client/mistAPIBackend.js]: dirname is added in the preloaded file for the mist interface. Though it might be smart to move change that do using `_dirname` as its anyway a module loaded in the preloaded file
Its a custom global variable i added. See: https://github.com/ethereum/mist/blob/develop/modules/preloader/mistUI.js#L33

Comment (hiddentao) [inline interface/client/mistAPIBackend.js]: That's interesting. I definitely have some problem with the preloader then - this may also be what caused my previous crash.

**Comment by hiddentao** [comment]
Shifted the code into a preloader now and changed where IPC listens to it. Much nicer now.

**Comment by hiddentao** [comment]
@bas-vk @alexvandesande Would be nice to get another pair of eyes on this.

File: interface/i18n/app.de.i18n.json [hunk 4]
```diff
@@ -1,34 +0,0 @@
...
-        "you": "Du",
-        "send": "Senden",
-        "or": "oder",
-        "of": "von",
-        "with": "mit",
-        "and": "und",
-        "on": "auf",
-        "per": "pro"
-    }
-}
```
Comment (alexvandesande) [inline interface/i18n/app.de.i18n.json]: Why did you delete these german translation files?

**Comment by alexvandesande** [comment]
I tried running your branch but got this error from electron:

```
App threw an error when running [SyntaxError: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode]
```
I ran `npm install` to check if it was a new dependency but the error persisted.

**Comment by hiddentao** [comment]
What version of node are you using?

File: interface/i18n/app.de.i18n.json (hunk 4 above)
Comment (hiddentao) [inline interface/i18n/app.de.i18n.json]: Sorry about that, fixed now.

**Comment by alexvandesande** [comment]
@hiddentao v0.12.3

**Comment by hiddentao** [comment]
Ah ok, so Node 4 is required for this. That's the current LTS release version of node and allows us to use the latest ES6 features. Will update package.json and the README.
